trestle/core/control_io.py

```diff
@@ -1096,39 +1116,115 @@ def read_new_alters_and_params(control_path: pathlib.Path) -> Tuple[List[prof.Al
                     new_alters.append(alter)
             else:
                 ii += 1
+        # this reads strings from the yaml as param values.  could convert to params
         header_params = header.get(const.SET_PARAMS_TAG, {})
         if header_params:
-            param_dict.update(header_params)
-        return new_alters, param_dict
+            param_str_dict.update(header_params)
+        return new_alters, param_str_dict
 
     @staticmethod
-    def get_control_param_dict(control: cat.Control,
-                               values_only: bool,
-                               params_format: Optional[str] = None) -> Dict[str, str]:
-        """Get a dict of the parameters in a control and their values."""
-        param_dict: Dict[str, str] = {}
-        params: List[common.Parameter] = as_list(control.params)
-        for param in params:
-            value_str = 'No value found'
-            if param.label:
-                value_str = param.label
-            if param.values:
-                values = [val.__root__ for val in param.values]
-                if len(values) == 1:
-                    value_str = values[0]
-                else:
-                    value_str = f"[{', '.join(value for value in values)}]"
-            # if there isn't an actual value then ignore this param
-            elif values_only:
-                continue
-            if params_format:
-                if params_format.count('.') > 1:
-                    raise TrestleError(
-                        f'Additional text {params_format} '
-                        f'for the parameters cannot contain multiple dots (.)'
-                    )
-                value_str = params_format.replace('.', value_str)
-            param_dict[param.id] = value_str
+    def param_values_as_str_list(param: common.Parameter) -> List[str]:
+        """Convert param values to list of strings."""
+        return [val.__root__ for val in as_list(param.values)]
+
+    @staticmethod
+    def param_values_as_str(param: common.Parameter, brackets=False) -> Optional[str]:
+        """Convert param values to string with optional brackets."""
+        if not param.values:
+            return None
+        values_str = ', '.join(ControlIOReader.param_values_as_str_list(param))
+        return f'[{values_str}]' if brackets else values_str
```

Comment:
Is there a need to keep those two methods separate? Having only one of them public would be preferable?

Reply:
Yes they are different use cases - a comma separated list as a single string, or an actual list of strings.